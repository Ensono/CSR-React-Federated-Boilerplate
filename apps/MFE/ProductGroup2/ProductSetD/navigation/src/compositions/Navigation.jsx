import React from "runtime-library/lib/react"
import AppBar from "runtime-library/lib/@mui/material/AppBar"
import Box from "runtime-library/lib/@mui/material/Box"
import Toolbar from "runtime-library/lib/@mui/material/Toolbar"
import Typography from "runtime-library/lib/@mui/material/Typography"
import IconButton from "runtime-library/lib/@mui/material/IconButton"
import Menu from "runtime-library/lib/@mui/material/Menu"
import MenuItem from "runtime-library/lib/@mui/material/MenuItem"
import MenuIcon from "runtime-library/lib/@mui/icons-material/Menu"
import Link from "runtime-library/lib/@mui/material/Link"

export function Navigation(props) {
  const [anchorElUser, setAnchorElUser] = React.useState(null)
  console.log("test")
  const links = [
    {
      name: "Induct Dashboard",
      url: "http://localhost:3004",
    },
    {
      name: "Sorter Dashboard",
      url: "http://localhost:3003",
    },
    {
      name: "Area Dashboard",
      url: "http://localhost:3002",
    },
    {
      name: "Efthy Dashboard",
      url: "http://localhost:3002",
    },
  ]
  const handleOpenUserMenu = event => {
    setAnchorElUser(event.currentTarget)
  }
  const handleCloseUserMenu = () => {
    setAnchorElUser(null)
  }
  return (
    <Box sx={{flexGrow: 1}}>
      <AppBar position="static" sx={{backgroundColor: "#333"}}>
        <Toolbar>
          <IconButton
            size="large"
            edge="start"
            color="inherit"
            aria-label="menu"
            sx={{mr: 2}}
            onClick={handleOpenUserMenu}
          >
            <MenuIcon />
          </IconButton>
          <Menu
            sx={{mt: "45px"}}
            id="menu-appbar"
            anchorEl={anchorElUser}
            anchorOrigin={{
              vertical: "top",
              horizontal: "right",
            }}
            keepMounted
            transformOrigin={{
              vertical: "top",
              horizontal: "right",
            }}
            open={Boolean(anchorElUser)}
            onClose={handleCloseUserMenu}
          >
            {links.map(link => (
              <MenuItem key={link.name} onClick={handleCloseUserMenu}>
                <Link href={link.url} underline="none" sx={{color: "#333"}}>
                  <Typography textAlign="center">{link.name}</Typography>
                </Link>
              </MenuItem>
            ))}
          </Menu>
          <Typography variant="h6" component="div" sx={{flexGrow: 1}}>
            {props.appName}
          </Typography>
          <Typography
            variant="h6"
            component="div"
            sx={{flexGrow: 1, textAlign: "right", fontSize: ".8rem"}}
          >
            Next Warehouse
          </Typography>
        </Toolbar>
      </AppBar>
    </Box>
  )
}

export default Navigation
