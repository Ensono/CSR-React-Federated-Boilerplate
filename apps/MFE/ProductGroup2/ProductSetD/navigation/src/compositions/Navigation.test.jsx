import React from "react"
import {render} from "@testing-library/react"
import Navigation from "./Navigation"

describe("navigation: ", () => {
  const props = {}

  it("Navigation should match the snapshot for type = primary", () => {
    const {asFragment} = render(<Navigation {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
