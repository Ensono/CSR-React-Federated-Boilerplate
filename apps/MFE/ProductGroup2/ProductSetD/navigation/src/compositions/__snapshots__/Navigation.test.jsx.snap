// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation:  Navigation should match the snapshot for type = primary 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-1 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: static;
  background-color: #1976d2;
  color: #fff;
  background-color: #333;
}

.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  min-height: 56px;
}

@media (min-width:600px) {
  .emotion-2 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:0px) {
  @media (orientation: landscape) {
    .emotion-2 {
      min-height: 48px;
    }
  }
}

@media (min-width:600px) {
  .emotion-2 {
    min-height: 64px;
  }
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  margin-left: -12px;
  color: inherit;
  padding: 12px;
  font-size: 1.75rem;
  margin-right: 16px;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: transparent;
  }
}

.emotion-3.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-5 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-6 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: right;
  font-size: .8rem;
}

<div
    class="MuiBox-root emotion-0"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic emotion-1"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular emotion-2"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge emotion-3"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-4"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root emotion-5"
          />
        </button>
        <div
          class="MuiTypography-root MuiTypography-h6 emotion-6"
        />
        <div
          class="MuiTypography-root MuiTypography-h6 emotion-7"
        >
          Next Warehouse
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
